src/Deku/Interpret.js

```diff
@@ -90,47 +254,57 @@ export const makeText_ = (runOnJust) => (tryHydration) => (maybe) => (a) => (sta
 				.item(0))
 		) {
 			var i = 0;
-			// if the length is one, that means that there is only a comment because
-			// the text is empty
-			// so we need to add a text node
-			if (dom.childNodes.length === 1) {
-				dom.prepend(document.createTextNode(""));
-			} else {
-				for (var i = 0; i < dom.childNodes.length; i++) {
-					if (
-						dom.childNodes[i].nodeType === 8 &&
-						dom.childNodes[i].nodeValue === ptr
-					) {
-						i = i - 1;
-						break;
+			for (; i < dom.childNodes.length; i++) {
+				const ptrSplit = ptr.split('@-@');
+				if (
+					dom.childNodes[i].nodeType === 8 &&
+					dom.childNodes[i].nodeValue === ptrSplit[0]
+				) {
+					i = i - 1;
+					var textWasBlank = i === -1;
+					var textWasBlankAfterDynBeacon = dom.childNodes[i].nodeType === 8;
```

Comment:
won't this line throw if `i === -1`?

Reply:
If it's at the start of the loop, meaning `var i = 0;`, then we rewind from `0` to `-1` the line before this check.

Reply:
right, but if i=-1, then `dom.childNodes[i]` will be `undefined`?

Reply:
riiiight fixed!